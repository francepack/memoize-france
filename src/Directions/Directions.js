import React from 'react';

export function Directions(props) {
  return(
    <div>
      {props.displayDirections &&
        <div className='directions-overlay'>
          <div className='directions-box'>
            <article className='directions-text'>
              <p>There is always more to learn about coding. Review will help ensure you remember knowledge picked up along the way. Choose a category to be quizzed on, and test your knowledge! Try to finish each subject area!</p> 
              <p>You can revist the questions that gave you trouble at the bottom of the quiz in the Review category.</p>
            </article>
            <button className='direction-close' onClick={props.toggleDirections}>Close</button>
          </div>
        </div>
      }
    </div>
  );
}