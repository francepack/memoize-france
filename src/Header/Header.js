import React, { Component } from 'react';
import Directions from  '../Directions/Directions';

export default class Header extends Component {
  constructor() {
    super();
    this.state = {
      displayDirections: false
    }
  }

  toggleDirections = () => {
    this.setState({ displayDirections: !this.state.displayDirections })
  }

  render() {
    return(
      <header className="header">
        <h1 className="title">codeQuiz</h1>
        <button className="direction-btn" onClick={this.toggleDirections}>Info</button>
        <p className="subtitle">Find a subject to study and test your knowledge</p>
        <Directions 
          displayDirections={this.state.displayDirections}
          toggleDirections={this.toggleDirections}
        />
      </header>
    )
  }
}
        