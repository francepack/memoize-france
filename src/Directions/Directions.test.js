import React from 'react';
import Directions from './Directions';
import Header from '../Header/Header'
import { shallow } from 'enzyme';

import {configure} from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

const toggleDirections = jest.fn();

describe('Directions', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Directions 
        toggleDirections={toggleDirections}
      />
    )
    wrapper.setProps({ displayDirections: true, selectedQuestion: {id:'1', category:'People', question:'Who is it?', options:['me', 'you', 'neither', 'both'], answer:'you', link:'http://plcaeholder.com'} });
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke toggleDirections when the direction-close button is clicked', () => {
    wrapper.find('.direction-close').simulate('click', { preventDefault: () => {}});
    expect(toggleDirections).toBeCalled(); 
  });

});  