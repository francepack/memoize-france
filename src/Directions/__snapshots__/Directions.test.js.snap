// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Directions should match snapshot 1`] = `
<div>
  <div
    className="directions-overlay"
  >
    <article
      className="directions-box"
    >
      <p>
        There is always more to learn about coding. Pick a category to be quizzed on, and test your knowledge. Try to conquer each subject area!
      </p>
      <p>
        You can revist the questions that gave you trouble at the bottom of the quiz.
      </p>
      <p>
        To update questions saved for review, or to restart a whole category, simply refresh the page.
      </p>
      <button
        className="direction-close"
        onClick={[MockFunction]}
      >
        close
      </button>
    </article>
  </div>
</div>
`;
